packages/data/tests/unit/adapters/elide-facts-test.ts

```diff
@@ -43,13 +68,24 @@ module('Unit | Adapter | elide facts', function(hooks) {
   test('dataQueryFromRequest', function(assert) {
     const adapter = this.owner.lookup('adapter:elide-facts');
     const queryStr = adapter.dataQueryFromRequest(TestRequest);
-    assert.deepEqual(
+    assert.equal(
       queryStr,
       '{"query":"{ table1 { edges { node { m1 m2 r d1 d2 } } } }"}',
       'dataQueryFromRequest returns the correct query string for the given request'
     );
   });
 
+  test('dataQueryFromRequestV2', function(assert) {
+    const adapter = this.owner.lookup('adapter:elide-facts');
+    const queryStr = adapter.dataQueryFromRequestV2(TestRequestV2);
+    assert.equal(
+      queryStr,
+      // QUnit forces us to escape all our escapes which are escapes of escapes for the request. It's messy
+      '{"query":"{ table1(filter: \\\\\\"d3=in=(v1,v2);d4=in=(v3,v4);d5!=null;time=ge=2015-01-03;time=lt=2015-01-04;m1=gt=0\\\\\\",sort: \\\\\\"d1\\\\\\",first: \\\\\\"10000\\\\\\") { edges { node { m1 m2 r(p: 123,as: a) d1 d2 } } } }"}',
+      'dataQueryFromRequestV2 returns the correct query string for the given request V2'
+    );
+  });
```

Comment:
We will need more tests around this since there is a bit of logic in this function.